### PR TITLE
feat: add useProgressStatus hook

### PR DESCRIPTION
**Pull request by andrey-canon**
## Description

This adds a custom hook that allow to get the progress state. This will take the information from `/api/course_home/progress/{courseId}` and will  return the following information

- `isComplete`:  Indicates if the user has completed the course(completion 100%)
-   `isPassing`: Indicates if the user  has passed the course, this depends on the grading configuration
-   `isGraded`:  if the course is graded, studio has an option to mark a course as graded or not, however that information is not returned by the API, therefore the criteria was to check if the course has grading policies
-   `gradePercent`:  the grade percentage reached by the user
-   `completePercent`:   completion percentage, this is calculate based on complete, incomplete and locked units

### CODE REVIEW

File: src/hooks/useProgressStatus/useProgressStatus.tsx [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+      // some calculation are performed in order to get the real value.
+      const summary = data.completion_summary;
+      const courseGrade = data.course_grade;
+      const total = summary.complete_count + summary.incomplete_count + summary.locked_count;
+
+      const isComplete = summary.incomplete_count === 0;
+      const completePercent = total ? summary.complete_count / total : 0;
+
+      const gradePercent = courseGrade.percent;
+      const isPassing = courseGrade.is_passing;
```
Comment (johanv26) [inline src/hooks/useProgressStatus/useProgressStatus.tsx:104]: Here I have to distinguish or discuss of this one and `user_has_passing_grade`/
image

Comment (andrey-canon) [reply src/hooks/useProgressStatus/useProgressStatus.tsx:104]: What is the question ?

**Comment by johanv26** [comment]
This component where would be located, and all views of MFE learning would have it waiting and listening ?

**Comment by andrey-canon** [comment]
> This component where would be located in learning MFE, and all views would have it waiting and listening ?

Where you need it,  I mean the implementation of this is out of the scope of this pr

File: src/hooks/useProgressStatus/useProgressStatus.tsx (hunk 1 above)
Comment (johanv26) [reply src/hooks/useProgressStatus/useProgressStatus.tsx:104]: Why use isPassing instead `user_has_passing_grade`?

**Review by johanv26 — APPROVED** [review]
I made a question of Why use isPassing instead user_has_passing_grade? 
Anyway I think both could reach the same result, for the others I approve

File: src/hooks/useProgressStatus/useProgressStatus.tsx (hunk 1 above)
Comment (andrey-canon) [reply src/hooks/useProgressStatus/useProgressStatus.tsx:104]: if you know the difference please tell, I just made the easiest  implementation for me

File: src/hooks/useProgressStatus/useProgressStatus.tsx [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+        },
+      });
+    };
+    const getOnFailure = (error: Error) => {
+      // On failure the error will be logged on the browser console and the state will be set to the default value.
+      // eslint-disable-next-line no-console
+      console.error('Unable to retrive progress data with error:', error.message);
+      dispatch({ type: 'DEFAULT' });
+    };
+    getProgressData(courseId, getOnSuccess, getOnFailure);
```
[the remark this answers is not in the document]
Comment (andrey-canon) [reply src/hooks/useProgressStatus/useProgressStatus.tsx:121]: if the getConfig fails, probably another thing will fail first  before this code

File: src/hooks/useProgressStatus/useProgressStatus.tsx (hunk 1 above)
Comment (johanv26) [reply src/hooks/useProgressStatus/useProgressStatus.tsx:104]: Is tricky `is_passing` aka `passed` https://github.com/openedx/edx-platform/blob/master/lms/djangoapps/course_home_api/progress/serializers.py#L19 is based on cutoffs:
https://github.com/openedx/edx-platform/blob/open-release/maple.master/lms/djangoapps/grades/course_grade.py#L330
But analyzing deeper they seems the same due 
https://github.com/openedx/edx-platform/blob/master/xmodule/course_block.py#L1268

**Pull request merged**